test/resource/attributes_test.exs

```diff
@@ -91,5 +91,27 @@ defmodule Ash.Test.Resource.AttributesTest do
         end
       )
     end
+
+    test "does not raise if you pass underscore in `name`" do
```

Comment:
I am not sure if this test is necessary or wanted in the final PR. I wasn't too sure what kind of a convention Ash has for tests, so I wanted to raise the discussion here.

Reply:
I think its great 👍🏻 makes it much more explanatory if this one doesn't fail and the other does, and vice versa.